src/ondil/estimators/online_mvdistreg.py

```diff
@@ -0,0 +1,1826 @@
+                xp=xp_asc,
+                fp=fp[:, i][::-1],
+            )[::-1]
+    return out
+
+
+def make_model_array(X, eq, fit_intercept):
+    n = X.shape[0]
+
+    # TODO: Check difference between np.array and list more explicitly?
+    if isinstance(eq, str) and (eq == "intercept"):
+        if not fit_intercept:
+            raise ValueError(
+                "fit_intercept[param] is false, but equation says intercept."
+            )
+        out = make_intercept(n_observations=n)
+    else:
+        if isinstance(eq, str) and (eq == "all"):
+            if isinstance(X, np.ndarray):
+                out = X
+            if HAS_PANDAS and isinstance(X, pd.DataFrame):
+                out = X.to_numpy()
+            if HAS_POLARS and isinstance(X, pl.DataFrame):
+                out = X.to_numpy()
+        elif isinstance(eq, np.ndarray) or isinstance(eq, list):
+            if isinstance(X, np.ndarray):
+                out = X[:, eq]
+            if HAS_PANDAS and isinstance(X, pd.DataFrame):
+                out = X.loc[:, eq]
+            if HAS_POLARS and isinstance(X, pl.DataFrame):
+                out = X.select(eq).to_numpy()
+        else:
+            raise ValueError("Did not understand equation. Please check.")
+
+        if fit_intercept:
+            out = np.hstack((make_intercept(n), out))
+
+    return out
+
+
+def get_J_from_equation(self, X: np.ndarray):
+    J = {}
+    for p in range(self.distribution.n_params):
+        if isinstance(self._equation[p], str):
+            if self._equation[p] == "all":
+                J[p] = X.shape[1] + int(self._fit_intercept[p])
+            if self._equation[p] == "intercept":
+                J[p] = 1
+        elif isinstance(self._equation[p], np.ndarray) or isinstance(
+            self._equation[p], list
+        ):
+            J[p] = len(self._equation[p]) + int(self._fit_intercept[p])
+        else:
+            raise ValueError("Something unexpected happened")
+    return J
+
+
+@nb.njit()
+def get_max_lambda(x_gram: np.ndarray, y_gram: np.ndarray, is_regularized: np.ndarray):
+    if np.all(is_regularized):
+        max_lambda = np.max(np.abs(y_gram))
+    elif np.sum(~is_regularized) == 1:
+        intercept = y_gram[is_regularized] / np.diag(x_gram)[~is_regularized]
+        max_lambda = np.max(
+            np.abs(y_gram.flatten() - x_gram[~is_regularized] * intercept)
+        )
+    else:
+        raise NotImplementedError("Currently not implemented")
+    return max_lambda
+
+
+# TODO: This should use the distribution.parameter_shape
+def get_adr_regularization_distance(d: int, parameter_shape: str):
+    if parameter_shape == ParameterShapes.LOWER_TRIANGULAR_MATRIX:
+        j, i = np.triu_indices(d, k=0)
+    else:
+        i, j = np.triu_indices(d, k=0)
+    distance = np.abs(i - j)
+    return distance
+
+
+# TODO: This should use the
+def get_low_rank_regularization_distance(d, r):
+    return np.concatenate([np.repeat(i + 1, d) for i in range(r)])
```

Comment:
Only used once. Just move the logic directly into the code. This function essentially has zero lines.